plasma-mode.md

```diff
@@ -0,0 +1,203 @@
+# Purpose
+
+This document describes the design of Plasma Mode for different DA Layers with minimal op-node & specs repo changes.
+Once ratified, we will implement the described design.
+
+# Problem Statement + Context
+
+Plasma Mode is a modification the OP Stack protocol where the data is posted to a dedicated DA layer instead of Ethereum. This is done to reduce the cost of operating the specific chain. To maintain security, we provide a mechanism to challenge commitments and make data available on Ethereum when there is a dispute over the commitment.
+
+
+We have the following requirements for Plasma Mode:
+- Data is posted to a DA layer instead of Ethereum
+- There is a DA Challenge which validates that the data is available
+- Optionally, the DA challenge process makes the data available on Ethereum
+- The op-node is able to mark L2 as finalized when commitments used to derive the L2 chain are no longer able to marked as invalid
+
+Additionally, we take the following development-process requirement:
+- The op-node and specification repo does not need to be modified for each new DA Layer
+
+The largest challenge to this developement process requirement is that DA Providers have different ways of committing to data and semantics of challenges. We need to create a design that generically supports a broad set of DA Providers without heavy modification or forking of the OP Stack.
+
+## High Level Overview of Plasma Mode
+
+Instead of posting data directly to L1, the batcher gives the data to a DA server which posts the data to a DA layer.
+The DA server returns a commiment to the data to the batcher & the batcher posts that commitment to L1.
+The op-node performs derivation with the modification that when it finds a commiment on L1, it asks its DA server to get the data associated with the commitment.
+Then the rest of the derivation pipeline functions as normal.
+Finalization is modified because a L2 block can only be finalized when the commitment that was used to derive it can no longer be challenged.
+Commitment finalization takes significantly longer in Plasma Mode because of the challenge period.
+Data from commitments is also used prior to knowing if the commitment is valid or not.
+
+# Proposed Solution
+
+We propose a generic commitment type which can be interpreted by a DA server to fetch & validate the data. Challenge commitments is done in per DA layer smart contract where the smart contract emits events about the status of the challenge. The op-node then interprets those events to handle commitment validity and finality.
```

Comment:
Does this mean that every integration will need to implement their own challenge contract?

Reply:
Yes this is correct. The simplest contract that DA layers would have to implement is validating a commitment against data (i.e. validate the KZG or Merkle Hash or Merkle Mountain Range, etc) on L1 & then hook that into the current DAC smart contract.